addon-manager/tests/src/test/java/org/jboss/forge/addon/manager/ui/ForgeUpdateDistributionCommandTest.java

```diff
@@ -0,0 +1,290 @@
+import org.xml.sax.SAXException;
+
+@RunWith(Arquillian.class)
+public class ForgeUpdateDistributionCommandTest
+{
+   private static String FORGE_OLD_VERSION = "2.0.0.Final";
+
+   @Deployment
+   @AddonDependencies({
+            @AddonDependency(name = "org.jboss.forge.furnace.container:cdi"),
+            @AddonDependency(name = "org.jboss.forge.addon:resources"),
+            @AddonDependency(name = "org.jboss.forge.addon:addon-manager"),
+            @AddonDependency(name = "org.jboss.forge.addon:dependencies"),
+            @AddonDependency(name = "org.jboss.forge.addon:maven"),
+            @AddonDependency(name = "org.jboss.forge.addon:ui-test-harness")
+   })
+   public static AddonArchive getDeployment()
+   {
+      AddonArchive archive = ShrinkWrap
+               .create(AddonArchive.class)
+               .addBeansXML()
+               .addAsResource(ForgeUpdateDistributionCommandTest.class.getResource(TEST_OLD_FORGE), TEST_OLD_FORGE);
+      return archive;
+   }
+
+   private static String TEST_OLD_FORGE = "/forge-old-distribution.zip";
+
+   @Inject
+   private Addon addon;
+
+   @Inject
+   private Furnace furnace;
+
+   @Inject
+   private UITestHarness uiTestHarness;
+
+   @Test
+   public void testUpdateDistributionCommand() throws Exception
+   {
+      File tempDir = OperatingSystemUtils.createTempDir();
+      File inputFile = File.createTempFile("forge-old-distribution", ".zip", tempDir);
+      inputFile.deleteOnExit();
+      try (InputStream iStream = getClass().getResourceAsStream(TEST_OLD_FORGE))
+      {
+         try (OutputStream oStream = new FileOutputStream(inputFile))
+         {
+            IOUtils.copy(iStream, oStream);
+         }
+      }
+
+      ZipFile zipFile = new ZipFile(inputFile.getAbsolutePath());
+      String extractedFolderPath = tempDir.getAbsolutePath() + "/extracted-forge";
+      new File(extractedFolderPath).mkdirs();
+      zipFile.extractAll(extractedFolderPath);
+      String forgeHome = extractedFolderPath + "/forge-old-distribution";
+      System.setProperty("forge.home", forgeHome);
+      File homeAddonsDir = new File(forgeHome + "/addons");
+      Assert.assertTrue(homeAddonsDir.exists());
+      Assert.assertTrue(homeAddonsDir.listFiles().length == 0);
+      String uiPreviousVersion = getPreviousVersion(addon.getRepository().getRootDirectory().getPath());
+      changeManagerAddonDirectoryToBeOlder(addon.getRepository().getRootDirectory(), uiPreviousVersion);
+      waitForAddonManagerAddon(furnace);
+      do
+      {
+         // we need to wait untill furnace will process all the information changed in the directories
+         Thread.sleep(500);
+      }
+      while (furnace.getAddonRegistry().getServices("org.jboss.forge.addon.manager.impl.ui.DistributionUpdateCommand")
+               .isUnsatisfied());
+
+      uiTestHarness.setGui(false);
+      try (CommandController controller = uiTestHarness.createCommandController("Forge Update Distribution"))
```

Comment:
The test method body should begin here. Everything else above it should be moved to a @Before method